### PR TITLE
fix(website): better footer + fix redirects + remove all trailing slashes

### PR DESCRIPTION
**Pull request by nahoc**
Solves: https://linear.app/risczero/issue/DR-330/all-dev-docs-search-include-a-10-prefix-leads-to-404

So here are the things that were causing this:
1. Algolia hadn't been re-indexed since sept 24th
2. Algolia was returning some URLs with trailing slashes because we had some links scattered here and there with trailing links -- to remedy this I added a remark plugin to remove trailing slashes everywhere and on save
3. URLs with `1.1` in them were returning 404, so I wrote a regex to redirects when the `latestVersion` is in the URL, so this should be dynamic and follow newer versions aswell in the future

I also update the footer with some more relevant links ✨ 

---

old:

image

---

new (mobile)

image

new (desktop)

image

### CODE REVIEW

**Comment by pdg744** [comment]
Agreed your post screenshot looks better than the pre screenshot! 

I took a moment to check what the footer looks like on my machine, and I'm seeing a less useful footer (on Safari & Firefox on my Mac, and on Safari on my iPhone). @nahoc any idea why I'm not seeing what you're showing in the screenshots above? 

image

**Comment by nahoc** [comment]
> Agreed your post screenshot looks better than the pre screenshot!
> 
> I took a moment to check what the footer looks like on my machine, and I'm seeing a less useful footer (on Safari & Firefox on my Mac, and on Safari on my iPhone). @nahoc any idea why I'm not seeing what you're showing in the screenshots above?
> 
> image

Have you pulled my branch properly? Seems like you just have the old footer showing there.. shouldn't matter what browser

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -75,25 +76,33 @@ export default async function createConfigAsync() {
...
-            if (path.includes("/api/zkvm")) {
-              return [path.replace("/api/zkvm", "/zkvm")];
+
+            if (existingPath.includes("/api/zkvm")) {
+              return [existingPath.replace("/api/zkvm", "/zkvm")];
+            }
+
+            if (existingPath.includes(`/api`)) {
+              return [existingPath.replace(`/api`, `/api/${apiVersions[0]}`)];
             }
```
Comment (nahoc) [inline website/docusaurus.config.js:99]: basically saying `/api/1.1/*` redirects to `/api/*`

**Comment by nuke-web3** [comment]
`bun i && bun run start`
```sh
[SUCCESS] Docusaurus website is running at: http://localhost:3000/

[ERROR] TypeError: Ajv is not a constructor
    at /home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/validate.js:31:15
    at /home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/util/memorize.js:20:44
    at validateObject (/home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/validate.js:169:26)
    at validate (/home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/validate.js:155:14)
    at new MiniCssExtractPlugin (/home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/dist/index.js:497:5)
    at createBaseConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/webpack/base.js:200:13)
    at async createBaseClientConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/webpack/client.js:24:24)
    at async createStartClientConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/webpack/client.js:51:55)
    at async getStartClientConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/commands/start/webpack.js:102:36)
    at async createWebpackDevServer (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/commands/start/webpack.js:116:20)
[INFO] Docusaurus version: 3.4.0
Node version: v20.14.0
error: script "start" exited with code 1 (SIGHUP)
```

**Comment by nahoc** [comment]
> `bun i && bun run start`
> 
> ```shell
> [SUCCESS] Docusaurus website is running at: http://localhost:3000/
> 
> [ERROR] TypeError: Ajv is not a constructor
>     at /home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/validate.js:31:15
>     at /home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/util/memorize.js:20:44
>     at validateObject (/home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/validate.js:169:26)
>     at validate (/home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/node_modules/schema-utils/dist/validate.js:155:14)
>     at new MiniCssExtractPlugin (/home/nuke/git/r0/risc0/website/node_modules/mini-css-extract-plugin/dist/index.js:497:5)
>     at createBaseConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/webpack/base.js:200:13)
>     at async createBaseClientConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/webpack/client.js:24:24)
>     at async createStartClientConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/webpack/client.js:51:55)
>     at async getStartClientConfig (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/commands/start/webpack.js:102:36)
>     at async createWebpackDevServer (/home/nuke/git/r0/risc0/website/node_modules/@docusaurus/core/lib/commands/start/webpack.js:116:20)
> [INFO] Docusaurus version: 3.4.0
> Node version: v20.14.0
> error: script "start" exited with code 1 (SIGHUP)
> ```
> 
> What might I have missing here @nahoc ? 😓

be sure bun is up-to-date (`bun upgrade`) and try to delete node_modules and re-install

**Review by flaub — APPROVED** [review]

File: website/package.json [hunk 2]
```diff
@@ -42,26 +39,27 @@
...
     "remark-preset-lint-consistent": "6.0.0",
     "remark-preset-lint-markdown-style-guide": "6.0.0",
     "remark-preset-lint-recommended": "7.0.0",
+    "remark-remove-url-trailing-slash": "2.0.0",
     "remark-sort-definitions": "2.0.0",
     "remark-toc": "9.0.0",
     "remark-wiki-link": "2.0.1"
   },
-  "packageManager": "bun@1.1.21"
+  "packageManager": "bun@1.1.29"
```
Comment (nuke-web3) [inline website/package.json:64]: https://github.com/renovatebot/renovate/issues/29941 seems like there should be ways to force this? this setting was not respected AFAICT

**Pull request merged**